aws/kms/key/key.go

```diff
@@ -0,0 +1,225 @@
+// Package key provides filtering of KMS keys by Description
+package key
+
+import (
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/GSA/grace-tftest/aws/kms/policy"
+	"github.com/GSA/grace-tftest/aws/shared"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/kms"
+)
+
+// Key contains the necessary properties for testing *kms.KeyMetadata objects
+type Key struct {
+	client  client.ConfigProvider
+	key     *kms.KeyMetadata
+	filters []shared.Filter
+}
+
+// New returns a new *Key
+func New(client client.ConfigProvider) *Key {
+	return &Key{client: client}
+}
+
+// Selected returns the currently selected *kms.KeyMetadata
+func (a *Key) Selected() *kms.KeyMetadata {
+	return a.key
+}
+
+// Key returns the currently selected Keys' targeted *kms.KeyMetadata
+func (a *Key) Key(t *testing.T) *kms.KeyMetadata {
+	if a.key == nil {
```

Comment:
should probably consistently use `a.Selected()` or `a.key` when checking whether a key has been selected.

Reply:
went with a.key